src/Illuminate/Database/Console/Factories/FactoryMakeCommand.php

```diff
@@ -48,7 +48,7 @@ protected function buildClass($name)
     {
         $namespaceModel = $this->option('model')
                         ? $this->qualifyClass($this->option('model'))
-                        : trim($this->rootNamespace(), '\\').'\\Model';
+                        : trim($this->rootNamespace(), '\\').'\\'. ucfirst(str_replace('Factory', '', $name));
```

Comment:
Replacing every occurrence of Factory is not what we want surely. What about a model actually being called `Factory`? Surely we should instead just delete the last 7 characters if they are `Factory`?

Reply:
Yeah good point !! what if there is no `Factory` and we deleted the model name i think we should check if the string has a `Factory` string ?